tools/language-server/astro.d.ts

```diff
@@ -0,0 +1,28 @@
+type AstroRenderedHTML = string;
+
+interface FetchContentResult {
+  astro: {
+    headers: string[];
+    source: string;
+    html: AstroRenderedHTML;
+  };
+  url: URL;
+  [key: string]: any;
+}
+
+interface AstroPageRequest {
+  url: URL;
+  canonicalURL: URL;
+}
+
+interface Astro {
+  isPage: boolean;
+  fetchContent(globStr: string): FetchContentResult[];
+  props: Record<string, number | string | any>;
```

Comment:
If `export interface Props` is defined, we should use `Props` here. Fine if that's a follow-up.

Reply:
Good idea, going to file an issue to do this in a follow-up. I'm not sure how to do that in a .d.ts file like this as we would presumably have to parse the frontmatter to see if a Props exists. Unless there's a TypeScript equivalent of `typeof Thing === 'undefined'` that I'm not aware of...

Reply:
Can you file an issue to discuss? Part of me feels like making Astro.props contextual isn't a great idea. Although I can definitely see the utility too. My concern would be that this wouldn't work with any other tools as this depends on some magic only in the vscode extension, where as `Astro.props as MyProps` is explicit. Definitely on the fence here and want to discuss though.

Reply:
> My concern would be that this wouldn't work with any other tools

If we think of this as a responsibility of the TS parser / plugin / whatever it is that allows TS to run on `.astro` files, then I think we can consider this as "works with all other tools".